git_machete/github.py

```diff
@@ -86,47 +86,54 @@ def from_url(cls,
             if match:
                 org = match.group(1)
                 repo = match.group(2)
-                return RemoteAndOrganizationAndRepository(remote=remote,
-                                                          organization=org,
-                                                          repository=repo if repo[-4:] != '.git' else repo[:-4])
-        return RemoteAndOrganizationAndRepository(remote=None,
-                                                  organization=None,
-                                                  repository=None)
+                return cls(remote=remote,
+                           organization=org,
+                           repository=repo if repo[-4:] != '.git' else repo[:-4])
+        return cls(remote=None,
+                   organization=None,
+                   repository=None)
 
 
 class GitHubToken:
     GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
 
     def __init__(self,
-                 domain: str
+                 value: Optional[str],
```

Comment:
Hmmm actually... shouldn't these properties be non-nullable tho? 🤔 is there any case where we'd want to pass `None` as either of these? See the original issue #819

Reply:
Yes, after the changes these values are non-nullable